crates/tracing/src/metrics.rs

```diff
@@ -0,0 +1,51 @@
+use heck::ToKebabCase;
+
+#[cfg(feature = "otel")]
+const DEFAULT_METRICS_ENDPOINT: &str = "http://localhost:4318/v1/metrics";
+
+#[cfg(not(feature = "otel"))]
+pub fn configure_metrics(_: String, _: &wasmcloud_core::OtelConfig) -> anyhow::Result<()> {
+    // no-op for when otel isn't enabled.
+    Ok(())
+}
+
+#[cfg(feature = "otel")]
+#[allow(clippy::missing_errors_doc)]
+#[allow(clippy::module_name_repetitions)]
+pub fn configure_metrics(
+    service_name: &str,
+    otel_config: &wasmcloud_core::OtelConfig,
+) -> anyhow::Result<()> {
+    use opentelemetry_otlp::WithExportConfig;
+    let normalized_service_name = service_name.to_kebab_case();
+
+    let endpoint = if let Some(ref endpoint) = otel_config.exporter_otlp_endpoint {
+        endpoint.clone()
+    } else {
+        eprintln!(
+            "OTEL exporter endpoint not set, defaulting metrics to '{DEFAULT_METRICS_ENDPOINT}'"
+        );
+        DEFAULT_METRICS_ENDPOINT.to_string()
+    };
+
+    let res = opentelemetry_otlp::new_pipeline()
+        .metrics(opentelemetry::runtime::Tokio)
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .http()
+                .with_endpoint(endpoint)
+                .with_protocol(opentelemetry_otlp::Protocol::HttpBinary),
+        )
+        .with_resource(opentelemetry::sdk::Resource::new(vec![
+            opentelemetry::KeyValue::new("service.name", normalized_service_name),
+        ]))
+        .with_aggregation_selector(
+            opentelemetry::sdk::metrics::reader::DefaultAggregationSelector::new(),
+        )
+        .with_temporality_selector(
+            opentelemetry::sdk::metrics::reader::DefaultTemporalitySelector::new(),
+        )
+        .build();
```

Comment:
I think the otel crate errors can be converted to anyhow errors, so you should be able to do the following and drop the `map_err` from the next line
```suggestion
        .build().context("failed to create OTEL metrics provider")?;
```

Reply:
Good callout, thank you 👍